internal/executor/reporter/event.go

```diff
@@ -137,8 +141,87 @@ func CreateJobIngressInfoEvent(pod *v1.Pod, clusterId string, associatedServices
 	}, nil
 }
 
-func CreateSimpleJobFailedEvent(pod *v1.Pod, reason string, clusterId string) api.Event {
-	return CreateJobFailedEvent(pod, reason, api.Cause_Error, []*api.ContainerStatus{}, map[string]int32{}, clusterId)
+func CreateJobPreemptedEvent(clusterEvent *v1.Event, clusterId string) (event *api.JobPreemptedEvent, err error) {
+	event = &api.JobPreemptedEvent{
+		ClusterId: clusterId,
+		Created:   clusterEvent.EventTime.Time,
+		Message:   clusterEvent.Message,
+	}
+
+	if err := enrichPreemptedEventFromInvolvedObject(event, clusterEvent.InvolvedObject); err != nil {
+		return nil, err
+	}
+
+	if clusterEvent.Related != nil {
+		if err := enrichPreemptedEventFromRelatedObject(event, clusterEvent.Related); err != nil {
+			return nil, err
+		}
+	}
+
+	if err := enrichPreemptedEventFromPreemptionMessage(event, clusterEvent.Message); err != nil {
+		return nil, err
+	}
+
+	return event, nil
+}
+
+func enrichPreemptedEventFromInvolvedObject(event *api.JobPreemptedEvent, involved v1.ObjectReference) error {
+	preemptedJobId, err := util.ExtractJobIdFromName(involved.Name)
+	if err != nil {
+		return errors.WithMessage(err, "error extracting preempted job id from pod name")
+	}
+
+	event.JobId = preemptedJobId
+	event.PreemptedPodNamespace = involved.Namespace
+	event.PreemptedPodName = involved.Name
+
+	return nil
+}
+
+func enrichPreemptedEventFromRelatedObject(event *api.JobPreemptedEvent, related *v1.ObjectReference) error {
+	if strings.HasPrefix(related.Name, common.PodNamePrefix) {
+		preemptiveJobId, err := util.ExtractJobIdFromName(related.Name)
+		if err != nil {
+			return errors.WithMessage(err, "error extracting preemptive job id from pod name")
+		}
+
+		event.PreemptiveJobId = preemptiveJobId
+	}
+
+	event.PreemptivePodNamespace = related.Namespace
+	event.PreemptivePodName = related.Name
+
+	return nil
+}
+
+func enrichPreemptedEventFromPreemptionMessage(event *api.JobPreemptedEvent, msg string) error {
+	info, err := util.ParsePreemptionMessage(msg)
+	if err != nil {
+		return err
+	}
+
+	if event.PreemptiveJobId == "" && util.IsArmadaJobPod(info.Name) {
+		jobId, err := util.ExtractJobIdFromName(info.Name)
+		if err != nil {
```

Comment:
should this be a warning? I'm not sure what action ends up getting taken here but it's probably better to generate an event with the `PreemptiveJobId ` missing than it is to lose the event completely.